### PR TITLE
1291 deprecate index artifact

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java [hunk 1]
```diff
@@ -48,13 +48,19 @@
...
      * @throws org.sleuthkit.datamodel.TskCoreException
      */
+    @Deprecated 
     public void indexArtifact(BlackboardArtifact artifact) throws TskCoreException;
 
     /**
-     * Add the given Content object to the text index.
+     * Add the given Content object to the text index.  This message should only
+     * be used in atypical cases, such as indexing a report.  Artifacts are indexed 
+     * when postArtifact called and files are indexed during ingest.
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java]: Missing an "is" between "postArtifact" and "called." I would also like to see postArtifact qualified with the class name. Perhaps "org.sleuthkit.autopsy.datamodel.Blackboard.postArtifact."

File: Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java [hunk 2]
```diff
@@ -48,13 +48,19 @@
      * all of its attributes.
      *
      * @param artifact The artifact to index.
+     * 
+     * @deprecated postArtifact org.sleuthkit.autopsy.datamodel.Blackboard will
+     * take care of this functionality at ingest
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java]: This seems a bit choppy: method name, class name. Perhaps simply say: "Call org.sleuthkit.autopsy.datamodel.Blackboard.postArtifact instead."

File: Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java [hunk 3]
```diff
@@ -48,13 +48,19 @@
      * all of its attributes.
      *
      * @param artifact The artifact to index.
+     * 
+     * @deprecated postArtifact org.sleuthkit.autopsy.datamodel.Blackboard will
```
Comment (bcarrier) [inline Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java]: this should reference the TSK datamodel, not Autopsy.

File: Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java (hunk 1 above)
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/keywordsearchservice/KeywordSearchService.java]: I would still like to see postArtifact qualified here.

**Pull request merged**